src/pk/ecc/ecc_recover_key.c

```diff
@@ -0,0 +1,222 @@
+/* LibTomCrypt, modular cryptographic library -- Tom St Denis
+ *
+ * LibTomCrypt is a library that provides various cryptographic
+ * algorithms in a highly modular and flexible manner.
+ *
+ * The library is free for all purposes without any express
+ * guarantee it works.
+ */
+
+#include "tomcrypt_private.h"
+
+#ifdef LTC_MECC
+
+#ifdef LTC_ECC_SHAMIR
+
+/**
+  @file ecc_recover_key.c
+  ECC Crypto, Russ Williams
+*/
+
+static int _ecc_recover_key(const unsigned char *sig,  unsigned long siglen,
+                            const unsigned char *hash, unsigned long hashlen,
+                            int recid, ecc_key *key)
+{
+   ecc_point     *mG = NULL, *mQ = NULL, *mR = NULL;
+   void          *p, *m, *a, *b;
+   void          *r, *s, *v, *w, *t1, *t2, *u1, *u2, *v1, *v2, *e, *x, *y;
+   void          *a_plus3 = NULL, *mu = NULL, *ma = NULL;
```

Comment:
shouldn't ` *a_plus3` be on the previous line (without `= NULL`)?

Reply:
It had the `= NULL` from `ecc_verify_hash()`, but I can't see a good reason for that. It's initialised with the rest in that giant `mp_init_multi()`, isn't optional (like ma/mu), and isn't a curve constant (like p/m/a/b).

Moved it and removed the NULL initialisation.